AmplifyPlugins/API/AWSAPICategoryPlugin/Interceptor/SubscriptionInterceptor/IAMAuthInterceptor.swift

```diff
@@ -94,61 +113,82 @@ class IAMAuthInterceptor: AuthInterceptor {
-         date: String,
+    init(host: String,
+         authorization: String,
+         securityToken: String,
+         amzDate: String,
          accept: String,
          contentEncoding: String,
-         contentType: String) {
-        self.date = date
+         contentType: String,
+         remainingHeaders: [String: String]?) {
         self.authorization = authorization
-        self.securityToken = token
+        self.securityToken = securityToken
+        self.amzDate = amzDate
         self.accept = accept
         self.contentEncoding = contentEncoding
         self.contentType = contentType
+        self.remainingHeaders = remainingHeaders
         super.init(host: host)
     }
 
-    private enum CodingKeys: String, CodingKey {
-        case authorization = "Authorization"
-        case accept
-        case contentEncoding = "content-encoding"
-        case contentType = "content-type"
-        case date = "x-amz-date"
-        case securityToken = "x-amz-security-token"
+    private struct DynamicCodingKeys: CodingKey {
+        var stringValue: String
+        init?(stringValue: String) {
+            self.stringValue = stringValue
+        }
+        var intValue: Int?
+        init?(intValue: Int) {
+            return nil
+        }
     }
 
     override func encode(to encoder: Encoder) throws {
-        var container = encoder.container(keyedBy: CodingKeys.self)
-        try container.encode(authorization, forKey: .authorization)
-        try container.encode(accept, forKey: .accept)
-        try container.encode(contentEncoding, forKey: .contentEncoding)
-        try container.encode(contentType, forKey: .contentType)
-        try container.encode(date, forKey: .date)
-        try container.encode(securityToken, forKey: .securityToken)
+        var container = encoder.container(keyedBy: DynamicCodingKeys.self)
+        try container.encode(authorization,
+                             forKey: DynamicCodingKeys(stringValue: SubscriptionConstants.authorizationkey)!)
+        try container.encode(securityToken,
+                             forKey: DynamicCodingKeys(stringValue: RealtimeProviderConstants.iamSecurityTokenKey)!)
+        try container.encode(amzDate,
+                             forKey: DynamicCodingKeys(stringValue: RealtimeProviderConstants.amzDate)!)
+        try container.encode(accept,
+                             forKey: DynamicCodingKeys(stringValue: RealtimeProviderConstants.acceptKey)!)
+        try container.encode(contentEncoding,
+                             forKey: DynamicCodingKeys(stringValue: RealtimeProviderConstants.contentEncodingKey)!)
+        try container.encode(contentType,
+                             forKey: DynamicCodingKeys(stringValue: RealtimeProviderConstants.contentTypeKey)!)
+        if let headers = remainingHeaders {
+            for (key, value) in headers {
+                try container.encode(value, forKey: DynamicCodingKeys(stringValue: key)!)
```

Comment:
looks like we never return `nil` as long as we instantiate `DynamicCodingKeys` using the string-based initializer, but I'd add a comment to help read and understand why force-unwrapping is relatively safe here